### PR TITLE
test(e2e): Switch from Appium to Maestro

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Enhancement
- [x] Refactoring

## :scroll: Description

This PR replaces appium and jest with maestro as runner of the e2e tests.

## :bulb: Motivation and Context

part of https://github.com/getsentry/projects/issues/165

## :green_heart: How did you test it?
ci, locally

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

#skip-changelog

### CODE REVIEW

**Review by antonis — COMMENTED** [review]
This is not related with the changes introduced in this PR but I noticed that the `Build RN , Upload logs` task always fails on Android ([example](https://github.com/getsentry/sentry-react-native/actions/runs/11575898278/job/32268389116?pr=4210))
```
Warning: No files were found with the provided path: dev-packages/e2e-tests/react-native-versions/0.76.0/RnDiffApp/ios/*.log. No artifacts will be uploaded.
```
I think replacing `ios` with the platform [on this line](https://github.com/getsentry/sentry-react-native/pull/4210/files#diff-3e103440521ada06efd263ae09b259e5507e4b8f7408308dc227621ad9efa31eR287) could fix this

`path: dev-packages/e2e-tests/react-native-versions/${{ matrix.rn-version }}/RnDiffApp/${{ matrix.platform }}/*.log`

File: dev-packages/e2e-tests/cli.mjs [hunk 1]
```diff
@@ -207,123 +206,40 @@
...
-      console.log('Appium server started');
-      return;
+      execSync('adb devices | grep -q "emulator"');
     } catch (error) {
-      console.log(`Appium server hasn't started yet (${error})...`);
-      await sleep(1000);
+      throw new Error('No Android emulator is currently running. Please start an emulator before running this script.');
     }
+
+    execSync(`adb install -r -d ${testApp}`);
```
[the remark this answers is not in the document]
Comment (krystofwoldrich) [reply dev-packages/e2e-tests/cli.mjs:231]: Good suggestion, but we control the input.

File: dev-packages/e2e-tests/cli.mjs [hunk 2]
```diff
@@ -207,123 +206,40 @@
...
+    } catch (error) {
+      throw new Error('No simulator is currently booted. Please boot a simulator before running this script.');
     }
-  }
-}
 
-async function waitForAppium() {
-  console.log('Waiting for Appium server to start...');
-  for (let i = 0; i < 60; i++) {
+    execSync(`xcrun simctl install booted ${testApp}`);
```
[the remark this answers is not in the document]
Comment (krystofwoldrich) [reply dev-packages/e2e-tests/cli.mjs:223]: Good suggestion, but we control the input.

**Comment by krystofwoldrich** [comment]
@antonis Good catch, we can just not run it for Android, as there are no logs generated. This should be for iOS only, because the xcodeoutput is saved to file.

**Review by antonis — APPROVED** [review]
LGTM 🚀

File: dev-packages/e2e-tests/cli.mjs (hunk 1 above)
[the remark this answers is not in the document]
Comment (krystofwoldrich) [reply dev-packages/e2e-tests/cli.mjs]: close

**Review by lucas-zimerman — APPROVED** [review]
LGTM 🚀

**Pull request merged**